nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-api/src/main/java/org/nuxeo/ecm/platform/comment/api/CommentManager.java

```diff
@@ -322,4 +323,36 @@ void deleteExternalComment(CoreSession session, String entityId)
      */
     boolean hasFeature(Feature feature);
 
+    /**
+     * Gets the top level ancestor document ref for the given document model comment ref. No matter how many levels of
+     * comments there is.
+     * <p>
+     * Given a document fileOne, that we comment with commentOne which we reply on with replyOne
+     * <p>
+     * This method will return:
+     * <ul>
+     * <li>CommentManager#getAncestorRef(session, commentOne) = fileOne</li>
+     * <li>CommentManager#getAncestorRef(session, replyOne) = fileOne</li>
+     * </ul>
+     *
+     * @param session the CoreSession
+     * @param commentIdRef the comment document model ref
+     * @return the top level ancestor document ref
+     * @since 11.1
+     */
+     default DocumentRef getAncestorRef(CoreSession session, DocumentRef commentIdRef){
+         throw new UnsupportedOperationException("Method not implemented !");
+     }
+
+    /**
+     * Gets the location where the comment will be stored.
```

Comment:
In addition of internal use (TreeCommentManager) and for instant this method is used [here](https://github.com/nuxeo/nuxeo/blob/1daebd3daee1ad241f98ce40f7233e961b6330ee/nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentsMigrator.java#L225) to avoid any duplication code or a cast to specific Manager when the migration is launched.

I don't if i answered you question

Reply:
To me, CommentMigrator usage is an internal usage as it is part of comment module.
What I meant is: this API is not meant to be used outside of comment service/migration purpose. We should state it in the javadoc.

Reply:
Now, i see what you mean. thx, i will add this in the javadoc